### PR TITLE
Handle more registers and range types in ex grammar

### LINKED ISSUE #1192: Missing some of Vim's mark-based features in ex-mode
# Issue type
- Bug report

# Environment
Emacs version: GNU Emacs 26.3
Operating System: Arch linux
Evil version: evil-git-3028d8e
Evil installation type: MELPA
Graphical/Terminal: X
Tested in a `make emacs` session: No

# Reproduction steps
From normal-mode, type
```:'[,'] some-command```
```:m '[```
```:,$ some-command```

where `some-command` is a valid ex command.

# Expected behavior
 some commonly used mark-based range features from Vim command-mode that seem to be missing:

1. Drop the `.` when referring to ranges. Vim lets me do
```:,$ s/what/ever/```

to mean the range `.,$`, current line to end of buffer. In evil I have to type the dot.

2. Marks as arguments to Ex commands. For instance, in Vim I can type
```:m '[```

To move the current line to where I last pasted text (which is the location `'[`).

3. For that matter ```:'[,'] some-command``` doesn't work either (such as ```:'[,'] !wc -w```)

# Actual behavior
Throws an error: `Invalid address` or `Ex: syntax error`

# Further notes
Perhaps these are implemented but disabled by default? I couldn't find them in the customization options.

Another possibility is these aren't available in vanilla Vim and are enabled by some plugin. I did test these on vim with ```vim -u NONE``` and they worked fine.

For some of these there are emacs commands (like `flush-lines` for `:%g/regexp/ d`) and `swiper` handles a few more, but in general ex commands are usually the fastest way to do one-time structured transformations that it doesn't make sense to write some elisp or define a macro for.

### PR DESCRIPTION
**Pull request by leungbk**
Fixes #1192.

I committed under @nnicandro's name. I think squash-merging them may incorrectly attribute authorship to me, whereas rebasing should correctly assign authorship to him.

### CODE REVIEW

**Comment by tomdl89** [comment]
Thanks for the PR @leungbk - looks good. Just had a couple of minor comments.

**Comment by leungbk** [comment]
What are your thoughts, @tomdl89?

**Review by tomdl89 — COMMENTED** [review]
@leungbk just these comments

**Comment by tomdl89** [comment]
Sorry, forgot to publish them

File: evil-ex.el [hunk 1]
```diff
@@ -67,13 +67,14 @@
...
      ("*" #'(evil-ex-last-visual-range))
+     ((alt "," ";") line #'(evil-ex-range (evil-ex-current-line) $2))
      (line ";" line #'(let ((tmp1 $1))
                         (save-excursion
                           (goto-line tmp1)
                           (evil-ex-range tmp1 $3))))
      (line "," line #'(evil-ex-range $1 $3))
      (line #'(evil-ex-range $1 nil))
-     ("`" "[-a-zA-Z_<>']" ",`" "[-a-zA-Z_<>']"
+     ("`" marker-name ",`" marker-name
```
[the remark this answers is not in the document]
Comment (leungbk) [reply evil-ex.el:77]: This was added a while ago as a bonus feature, it seems: https://github.com/emacs-evil/evil/issues/322#issuecomment-273749949

**Pull request merged**